toolchain/check/declaration_name_stack.h

```diff
@@ -11,6 +11,19 @@
 
 namespace Carbon::Check {
 
+// An index for a pushed scope, which may be a permanent scope with a
+// corresponding `NameScope`, or a temporary scope, such as is created for a
+// block.
```

Comment:
Fixed. (I moved the remaining comma earlier to be clearer what the "such as" binds to.)

Reply:
I still think the comma before "or" is incorrect, grammatically. It suggests they're two independent clauses, which is when you'd typically add a comma when there's not a third item (which, given the length of both halves, is part of my difficulty in reading -- I want to use the comma to separate clauses).

This might read better with "may be" -> "may either be", emphasizing the binary options.

Reply:
Found some more things I wanted to say here and split this to two sentences. I think that should resolve the issue.